tt_eager/tt_dnn/op_library/bcast/kernels/dataflow/reader_bcast_hw_interleaved.cpp

```diff
@@ -33,14 +36,19 @@ void kernel_main() {
     uint32_t l1_write_addr_in1;
 
     uint32_t num_tiles = src0_num_tiles;
-    uint32_t i = 0;
     uint32_t i1 = 0;
 
+    #ifndef IN0_SHARDED
```

Comment:
I prefer "if constexpr" to #ifdef
You can do constexpr based on src0_is_dram

Reply:
Wouldn't break this as #ifndef IN0_SHARDED is a pattern used across the kernels